wpilibj/src/main/java/edu/wpi/first/wpilibj/Timer.java

```diff
@@ -100,6 +100,18 @@ public void start() {
     }
   }
 
+  /**
+   * Restart the timer by stopping the timer, if it is not already stopped, then resting the
+   * internal counter, then starting the timer gain.
```

Comment:
```suggestion
   * internal counter, then starting the timer again.
```

Reply:
Maybe also use "accumulated time" instead of "internal counter"?

Reply:
We might also want to say something about the fact that `advanceIfElapsed` is a better choice in many repeating timer use cases, as it avoids timer drift.

Reply:
I'm not sure exactly what you mean, I have personally never used that method which is why I am a little confused. What would you want me to add to the docs?

Reply:
Essentially advanceIfElapsed enables this:
```
Timer timer;
timer.start();
while (true) {
  if (timer.advanceIfElapsed(1.0)) {
    // do something every 1.0 seconds
  }
  // sleep, do other stuff, etc.
}
```

This is approximately equivalent to:
```
while (true) {
  if (timer.get() >= 1.0) {
    timer.reset();
    // do something every 1.0 seconds
  }
  // sleep, do other stuff, etc.
}
```

Except the latter is subject to drift--it's not actually going to occur exactly every 1.0 seconds (relative to the start of the loop), it's going to be more than 1.0 seconds, as the get() might return say 1.05, but the reset() will reset to 0 (plus there's a bit of time that elapses between the get() and the reset()).

I don't know if we should add all this to the docs, but maybe something saying that "Instead of using this function, consider using advanceIfElapsed() if you want an event to periodically reoccur at some time interval from the start time."

Reply:
I wouldn't see someone using this restart method compared to that as this literally restarts the timer. Anyone who used it to achieve that goal would just be using it incorrectly (essentially just as reset method)

Reply:
> I have personally never used that method which is why I am a little confused.

Say you're using a timer to periodically actuate something every 10 ms. One way to do that is the following toy example:
```java
public class Robot extends TimedRobot {
  private Timer timer = new Timer();
  private Solenoid solenoid = new Solenoid(0);

  @Override
  public void teleopInit() {
    timer.start();
  }

  @Override
  public void teleopPeriodic() {
    if (timer.get() > 0.01) {
      timer.restart();
      solenoid.toggle();
    }
  }
}
```
`restart()` introduces a small delay between stopping and starting the timer again, so the start time gradually drifts from the TimedRobot periodic start time. `advanceIfElapsed()` doesn't have that problem.

Reply:
If it's that easy to use incorrectly, we should add documentation saying not to do what they're doing.

Reply:
What should I add to the documentation?

Reply:
"If you want an event to periodically reoccur at some time interval from the start time, consider using advanceIfElapsed() instead."